src/compiler/utilities.ts

```diff
@@ -11360,18 +11360,23 @@ export function createNameResolver({
                         location = root.parent;
                     }
                     break;
-                case SyntaxKind.Parameter:
+                case SyntaxKind.Parameter: {
+                    const param = location as ParameterDeclaration;
+                    if (lastLocation === param.name) {

```

Comment:
what does this do when param.name is a BindingPattern? Do we have tests to make sure that they still correctly get marked unused (and used when used)?

Reply:
yeah, there are tests for that already and the changes to `checkIfExpressionRefinesAnyParameter` that introduced this bug don't affect binding patterns